pkg/config/templateComponent.go

```diff
@@ -64,6 +65,73 @@ type TemplateData struct {
 	Data   []any
 }
 
+type ComponentStyle string
```

Comment:
ComponentType is the way the components were grouped into base, meta and template components.

Style is how we want the component to be styled in the UI. Style is a freeform text value right now, but could be made into an number like component group?

For example:
- receiver
- processor
- sampler
- exporter
- etc